pkg/git/remote_git_resolver.go

```diff
@@ -48,16 +78,65 @@ func (*remoteGitResolver) Resolve(auth transport.AuthMethod, sourceConfig corev1
 			}, nil
 		}
 	}
+	return corev1alpha1.ResolvedSourceConfig{}, err
+}
+
+func resolveSourceWithSubpath(auth transport.AuthMethod, sourceConfig corev1alpha1.SourceConfig) (corev1alpha1.ResolvedSourceConfig, error) {
+	var resolvedConfig corev1alpha1.ResolvedSourceConfig
+
+	r, err := gogit.Clone(memory.NewStorage(), nil, &gogit.CloneOptions{
+		URL:  sourceConfig.Git.URL,
+		Auth: auth,
+	})
+
+	rIter, _ := r.References()
+
+	rIter.ForEach(func(ref *plumbing.Reference) error {
+		if ref.Name().Short() == sourceConfig.Git.Revision {
+			refSourceType := sourceType(ref)
+			effectiveCommit := ""
+
+			if refSourceType == corev1alpha1.Branch {
+				effectiveCommit, err = latestCommitForSubpath(r, sourceConfig.SubPath)
```

Comment:
maybe we should do a `path.Clean(sourceConfig.SubPath)` or something when the values are compared to normalize the user provided subpath to handle the case that @chenbh mentioned where currently the use can provide `./subpath` or `subpath` and these may not both match correctly even though they are functionally equivalent. It may also make sense to normalize the paths provided by gogit to ensure they match what we expect. Everything else lgtm

Reply:
I've thought about this some more and I actually think we shouldn't do this. Supporting `.` can be taken as we also support `..` (because otherwise why use relative paths at all?) which is definitely not true. `path.Clean` might be counterproductive here because it'll actually processes `..` properly.

Reply:
I think the issue is that we support ./ today so we shouldn't break existing users. If someone uses ../ then that's fine. It probably won't work but it will be the same as before